aten/src/ATen/native/cpu/group_norm_kernel.cpp

```diff
@@ -798,21 +1323,43 @@ void GroupNormBackwardKernelImpl(
     Tensor& dX,
     Tensor& dgamma,
     Tensor& dbeta) {
+  // In training, using Amp to enable BFloat16 is recommended.
+  // It will keep module parameters in acc dtype i.e. float
+  // while input/output will be in BFloat16.
+  // Using parameters in BFloat16 will cause high precision loss.
   const bool mixed_type = is_mixed_type(dY, gamma);
-  AT_DISPATCH_FLOATING_TYPES_AND(
-      ScalarType::BFloat16, X.scalar_type(), "GroupNormBackwardKernelImpl", [&]() {
-      // In training, using Amp to enable BFloat16 is recommended.
-      // It will keep module parameters in acc dtype i.e. float
-      // while input/output will be in BFloat16.
-      // Using parameters in BFloat16 will cause high precision loss.
-      if(mixed_type) {
-        GroupNormBackwardKernelImplInternal<BFloat16, float>(
-            dY, X, mean, rstd, gamma, N, C, HxW, group, dX, dgamma, dbeta);
-      } else {
-        GroupNormBackwardKernelImplInternal<scalar_t, scalar_t>(
-            dY, X, mean, rstd, gamma, N, C, HxW, group, dX, dgamma, dbeta);
-      }
+  switch (X.suggest_memory_format()) {
+    case at::MemoryFormat::Contiguous: {
+      AT_DISPATCH_FLOATING_TYPES_AND(
+        ScalarType::BFloat16, X.scalar_type(), "GroupNormBackwardKernelImpl", [&]() {
+        if(mixed_type) {
+          GroupNormBackwardKernelImplInternal<BFloat16, float>(
+              dY, X, mean, rstd, gamma, N, C, HxW, group, dX, dgamma, dbeta);
+        } else {
+          GroupNormBackwardKernelImplInternal<scalar_t, scalar_t>(
+              dY, X, mean, rstd, gamma, N, C, HxW, group, dX, dgamma, dbeta);
+        }
```

Comment:
Same feedback as for other https://github.com/pytorch/pytorch/pull/81852 : can we somehow restrict this if to BFloat16 only (unless we can have a mixed precision case for float and double)

Reply:
This is bfloat16 only now, but float16 might be also added in the future. Use param_t = at::opmath_type<scalar_t> and GroupNormBackwardKernelImplInternal<scalar_t, param_t>for further  fp16 implementation.